src/main/java/io/github/noeppi_noeppi/mods/bongo/EventListener.java

```diff
@@ -193,35 +191,28 @@ public void resourcesReloaded(OnDatapackSyncEvent event) {
 
     @SubscribeEvent
     public void damage(LivingHurtEvent event) {
-        if (!event.getEntityLiving().getCommandSenderWorld().isClientSide && event.getEntityLiving() instanceof Player player && !event.getSource().isBypassInvul()) {
-            Bongo bongo = Bongo.get(player.getCommandSenderWorld());
-            Team team = bongo.getTeam(player);
-            if (bongo.running() && team != null) {
-                if (event.getSource().getEntity() instanceof Player source) {
-                    if (!bongo.getSettings().pvp) {
-                        event.setCanceled(true);
-                    } else if (team.hasPlayer(source)) {
-                        if (!bongo.getSettings().friendlyFire) {
-                            event.setCanceled(true);
-                        }
-                    }
-                } else if (bongo.getSettings().invulnerable) {
-                    event.setCanceled(true);
-                }
-            }
-        }
+        handleCommonDamageEvent(event, event.getSource(), () -> event.setAmount(0));
     }
 
     @SubscribeEvent
     public void attack(LivingAttackEvent event) {
-        if (!event.getEntityLiving().getCommandSenderWorld().isClientSide && event.getEntityLiving() instanceof Player player && !event.getSource().isBypassInvul()) {
+        handleCommonDamageEvent(event, event.getSource(), () -> {}); //can't modify damage
+    }
+
+    private void handleCommonDamageEvent(LivingEvent event, DamageSource source, Runnable setDamageToZero) {
+        if (event.getEntityLiving() instanceof Player p && p.isDamageSourceBlocked(source)) {
+            setDamageToZero.run();
```

Comment:
This will now always be called, even if `invulnerable` is turned off.

Reply:
it will always be called, if the damage should be blocked. If the damage is blocked, the damage should not apply and therefore it's valid to set the damage to 0

Reply:
In vanilla yes, but previous handlers may have changed the value and bongo would then break their behaviour.